src/main/java/duke/calendar/Date.java

```diff
@@ -0,0 +1,147 @@
+package duke.calendar;
+
+import duke.exception.DukeException;
+
+import java.time.Month;
+
+/**
+ * Represents a date. A <code>Date</code> object corresponds to a time represented by a day, month and year.
+ */
+public class Date {
+    protected String rawDate;
+    protected int day;
+    protected int year;
+    protected Month month;
+    protected boolean isNull = true;
+
+    /**
+     * Constructor for <code>Date</code>.
+     * @param rawDate Unprocessed date.
+     * @throws DukeException If provided date is invalid.
+     */
+    public Date(String rawDate) throws DukeException {
+        this.rawDate = rawDate;
+        if (rawDate != null) {
+            isNull = false;
+            processDate();
+        }
+    }
+
+    /**
+     * Extracts information about the day, month and year from the raw date.
+     * @throws DukeException If date is in the wrong format or invalid.
+     */
+    protected void processDate() throws DukeException {
+        String[] dateComponents = rawDate.split("/");
+        if (dateComponents.length < 3) {
+            throw new DukeException("☹ OOPS!!! Please specify a date in the form day/month/year. E.g. 2/12/2019");
+        } else {
+            int inputDay = Integer.parseInt(dateComponents[0]);
+            int inputYear = Integer.parseInt(dateComponents[2]);
+            int inputMonthNum = Integer.parseInt(dateComponents[1]);
+            year = inputYear;
+            if (isValidMonth(inputMonthNum)) {
+                month = Month.of(inputMonthNum);
+            } else {
+                throw new DukeException("☹ OOPS!!! Please specify a valid month.");
+            }
+            if (isValidDay(inputDay, inputMonthNum, inputYear)) {
+                day = inputDay;
+            } else {
+                throw new DukeException("☹ OOPS!!! Please specify a valid day.");
+            }
+        }
+    }
+
+    /**
+     * Checks if provided month is valid.
+     * @param inputMonthNum Number corresponding to specific month. E.g. 1 for January, 2 for February, etc.
+     * @return True if 1 <= inputMonthNum <= 12
+     */
+    protected boolean isValidMonth(int inputMonthNum) {
+        if (inputMonthNum < 1 || inputMonthNum > 12) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Checks if provided day is valid.
+     * @param inputDay Number corresponding to specific day
+     * @param inputMonthNum Number corresponding to specific month. E.g. 1 for January, 2 for February, etc.
+     * @param inputYear Number corresponding to year.
+     * @return True if day is valid for the given month and year.
+     */
+    protected boolean isValidDay(int inputDay, int inputMonthNum, int inputYear) {
+        switch (inputMonthNum) {
```

Comment:
What's the purpose of allowing the case to flow? Maybe there is a better way to refactor this portion to better reflect the purpose of this method

Reply:
You can use SimpleDateFormat instead to parse the String for a Date object, skipping all these.